tests/unittest_python3.py

```diff
@@ -298,6 +298,28 @@ def test_unpacking_in_dict_getitem(self) -> None:
             self.assertIsInstance(value, nodes.Const)
             self.assertEqual(value.value, expected)
 
+    def test_unpacking_in_dict_getitem_with_ref(self) -> None:
+        assign = extract_node(
+            """
+        a = {1: 2}
+        b = {**a, 2: 3}
+        """
+        )
+        node = assign.value
+
+        for key, expected in ((1, 2), (2, 3)):
+            value = node.getitem(nodes.Const(key))
+            self.assertEqual(value.value, expected)
+
+    def test_unpacking_in_dict_getitem_uninferable(self) -> None:
+        node = extract_node("{**a, 2: 3}")
```

Comment:
We can exercise the `not hasattr(..., "getitem")` with something like a lambda:
```suggestion
        node = extract_node(
            """
        a = lambda x: x
        {**a, 2: 3}
        """
        )
```

Reply:
If you prefer not to mix realistic cases with pathological cases, they could be separated.

Reply:
I changed the `hasattr` with an `isinstance` call. That should already be covered by the second test.